cmd/registry/controller/list.go

```diff
@@ -50,3 +55,47 @@ func ListResources(ctx context.Context, client connection.Client, pattern, filte
 
 	return result, nil
 }
+
+func generateApiHandler(result *[]ResourceInstance) func(*rpc.Api) {
+	return func(api *rpc.Api) {
+		apiName, _ := names.ParseApi(api.GetName())
+		resource := ApiResource{
+			ApiName:         ApiName{Api: apiName},
+			UpdateTimestamp: api.UpdateTime.AsTime(),
+		}
+		(*result) = append((*result), resource)
+	}
+}
+
+func generateVersionHandler(result *[]ResourceInstance) func(*rpc.ApiVersion) {
+	return func(version *rpc.ApiVersion) {
+		versionName, _ := names.ParseVersion(version.GetName())
+		resource := VersionResource{
+			VersionName:     VersionName{Version: versionName},
+			UpdateTimestamp: version.UpdateTime.AsTime(),
+		}
+		(*result) = append((*result), resource)
+	}
+}
+
+func generateSpecHandler(result *[]ResourceInstance) func(*rpc.ApiSpec) {
+	return func(spec *rpc.ApiSpec) {
+		specName, _ := names.ParseSpec(spec.GetName())
```

Comment:
Since we are passing valid rpc objects to these handlers (retrieved through the list response), there won't be a case of invalid specNames, hence ignoring the error here. The only reason to call `names.ParseSpec()` is to convert `rpc.Spec` object into `names.Name` object. Makes sense?

Reply:
Right, it should never happen. What should happen if we accidentally introduce a bug in the future that results in an error here? I think calling `panic(err)` might be more appropriate than proceeding as if the error didn't occur.

Reply:
Thanks for pushing for this.